src/lakefs_spec/spec.py

```diff
@@ -442,6 +640,31 @@ def __init__(
         size: int | None = None,
         **kwargs: Any,
     ):
+        """
+        Initialize a lakeFS file (local buffer on disk) for the given remote path for up- or downloads depending on ``mode``.
+
+        Parameters
+        ----------
+        fs: LakeFSFileSystem
+            The lakeFS file system associated to this file.
+        path: str
+            The remote path to either up- or download depending on ``mode``. Must be a fully qualified lakeFS URI.
+        mode: Literal["rb", "wb"]
+            The file mode indicating its purpose. Use ``rb`` for downloads from lakeFS, ``wb`` for uploads to lakeFS.
+        block_size: int | str
+            The file block size to read at a time. If not set, falls back to fsspec's default blocksize of 5 MB.
+        autocommit: bool
+            Whether to write the file buffer automatically to lakeFS on file closing in write mode.
+        cache_type: str
+            Cache type to use in the ``AbstractBufferedFile`` superclass.
+        cache_options: dict[str, Any] | None
+            Additional caching options to pass to the ``AbstractBufferedFile`` superclass.
+        size: int | None
+            If given and ``mode == 'rb'``, this will be used as the file size (in bytes).
+        kwargs: Any
+            Additional keyword arguments to pass to ``LakeFSClient.objects_api.get_object()`` on download (``mode == 'rb'),
+            or ``LakeFSClient.objects_api.put_object()`` on upload (``mode == 'wb').
+        """
```

Comment:
The Numpy doc style puts the constructor documentation into the class docstring: https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes

    The constructor (__init__) should also be documented here, the [Parameters](https://numpydoc.readthedocs.io/en/latest/format.html#params) section of the docstring details the constructor’s parameters.